### PR TITLE
Fix eslint and format scripts

### PR DESCRIPTION
**Pull request by iobuhov**
### Description

This PR introduce two new packages:

- `@mendix/eslint-config-web-widgets`
- `@mendix/prettier-config-web-widgets`

This two packages will be served as a source of shared configuration for `prettier` and `eslint`.
The problem with current approach is that we have eslint and prettier configs at the monorepo root and it not always plays well with source files we working on. So, by removing root configs and adding configs individually for each packages we: a) make packages more independent and b) we solving issues wit IDE.

Also in this PR I removed most of the dependencies from root `packages.json`. Now only `turbo` and `husky` are root deps. The rest is moved to `automation/scripts`.

Unfortunately this change require some configuration in your IDE.
If you use VS Code and you use official ESLint extension, then you have to add couple of lines to your `settings.json`

here is config:

```json
  "eslint.workingDirectories": [
    "./packages/tools/custom-widgets-utils-internal",
    "./packages/shared/prettier-config-web-widgets",
    "./packages/shared/charts",
    "./packages/shared/charts/hooks",
    "./packages/shared/charts/rollup",
    "./packages/shared/pluggable-widgets-commons/components/web",
    "./packages/shared/pluggable-widgets-commons",
    "./packages/shared/eslint-config-web-widgets",
    "./packages/pluggableWidgets/carousel-web",
    "./packages/pluggableWidgets/barcode-scanner-web",
    "./packages/pluggableWidgets/fieldset-web",
    "./packages/pluggableWidgets/image-web",
    "./packages/pluggableWidgets/maps-web",
    "./packages/pluggableWidgets/dropdown-sort-web",
    "./packages/pluggableWidgets/progress-circle-web",
    "./packages/pluggableWidgets/badge-web",
    "./packages/pluggableWidgets/color-picker-web",
    "./packages/pluggableWidgets/progress-bar-web",
    "./packages/pluggableWidgets/rating-web",
    "./packages/pluggableWidgets/pie-doughnut-chart-web",
    "./packages/pluggableWidgets/datagrid-web",
    "./packages/pluggableWidgets/html-element-web",
    "./packages/pluggableWidgets/time-series-chart-web",
    "./packages/pluggableWidgets/line-chart-web",
    "./packages/pluggableWidgets/video-player-web",
    "./packages/pluggableWidgets/accessibility-helper-web",
    "./packages/pluggableWidgets/language-selector-web",
    "./packages/pluggableWidgets/switch-web",
    "./packages/pluggableWidgets/timeline-web",
    "./packages/pluggableWidgets/tooltip-web",
    "./packages/pluggableWidgets/badge-button-web",
    "./packages/pluggableWidgets/accordion-web",
    "./packages/pluggableWidgets/datagrid-date-filter-web",
    "./packages/pluggableWidgets/datagrid-dropdown-filter-web",
    "./packages/pluggableWidgets/rich-text-web",
    "./packages/pluggableWidgets/heatmap-chart-web",
    "./packages/pluggableWidgets/range-slider-web",
    "./packages/pluggableWidgets/charts-web",
    "./packages/pluggableWidgets/popup-menu-web",
    "./packages/pluggableWidgets/column-chart-web",
    "./packages/pluggableWidgets/slider-web",
    "./packages/pluggableWidgets/datagrid-text-filter-web",
    "./packages/pluggableWidgets/bar-chart-web",
    "./packages/pluggableWidgets/datagrid-number-filter-web",
    "./packages/pluggableWidgets/bubble-chart-web",
    "./packages/pluggableWidgets/gallery-web",
    "./packages/pluggableWidgets/area-chart-web",
    "./packages/pluggableWidgets/tree-node-web",
    "./packages/jsActions/nanoflow-actions-hybrid/tests/phonegap",
    "./packages/modules/data-widgets",
    "./packages/modules/web-actions",
    "./packages/customWidgets/signature-web",
    "./packages/customWidgets/calendar-web",
    "./automation/utils",
    "./automation/scripts",
    "./automation/run-e2e"
  ]

```

### Pull request checklist

-   [x] All new and existing tests passed
-   [x] I run `lint` command locally and it doesn’t give errors
-   [ ] PR title properly formatted `[XX-000]: description`
-   [ ] Added record to packages' CHANGELOG.md
-   [ ] Bumped package version in `package.json` and `package.xml`
-   [ ] Added a link to related project PRs (atlas, pluggable-widgets-tools, testProject, etc.) (optional)
-   [ ] Created docs PR to [mendix/docs](https://github.com/mendix/docs.git) and added a link (optional)

### Pull request type

-   [ ] No code changes (changes to documentation, CI, metadata, etc)
-   [x] Dependency changes (any modification to dependencies in `package.json`)
-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)
-   [ ] Test related change (New E2E test, test automation, etc.)

### CODE REVIEW

File: .pnpmfile.cjs [hunk 1]
```diff
@@ -3,6 +3,11 @@ function readPackage(pkg, context) {
         delete pkg.dependencies["react-native"];
     }
 
+    if (pkg && pkg.name === "react-big-calendar") {
```
Comment (rahmanunver) [inline .pnpmfile.cjs:6]: why do we set the versions conditionally?

File: packages/customWidgets/calendar-web/tsconfig.json [hunk 2]
```diff
@@ -28,6 +28,5 @@
         "baseUrl": ".",
         "rootDir": "./",
         "outDir": "dist/tsc/"
-    },
-    "exclude": ["dist/", "tests/", "**/*.spec.ts", "**/*.specs.ts", "node_modules/"]
```
Comment (rahmanunver) [inline packages/customWidgets/calendar-web/tsconfig.json:32]: I get not excluding the specs, but node_modules?

File: .pnpmfile.cjs (hunk 1 above)
Comment (iobuhov) [reply .pnpmfile.cjs:6]: We just use condition to find package we want to fix (in our case `react-big-calendr`) and then we override `peerDependencies` to fix `pnpm` warning

File: packages/customWidgets/calendar-web/tsconfig.json (hunk 2 above)
Comment (iobuhov) [reply packages/customWidgets/calendar-web/tsconfig.json:32]: `typescript` will take care of this, we limit files by `"includes"` property, so `node_modules` will be not included.

**Review by rahmanunver — APPROVED** [review]

**Review by leonardomendix — APPROVED** [review]

**Review by rahmanunver — APPROVED** [review]

**Pull request merged**